dvc/repo/stage.py

```diff
@@ -371,13 +372,33 @@ def collect_repo(self, onerror: Callable[[str, Exception], None] = None):
                 the collection.
         """
         from dvc.dvcfile import is_valid_filename
+        from dvc.tree.local import LocalTree
 
-        stages = []
+        scm = self.repo.scm
+        sep = os.sep
         outs: Set[str] = set()
+
+        is_local_tree = isinstance(self.tree, LocalTree)
+
+        def is_ignored(path):
+            # apply only for the local tree
+            return is_local_tree and scm.is_ignored(path)
```

Comment:
This check might not work everywhere, especially in `for_write=True` scenarios in `dvcx`. But as we are adjusting our APIs in `ExternalRepo`, `Repo.clone()` should set some kind of state to workaround that.